src/skeleton/read.rs

```diff
@@ -39,27 +40,29 @@ pub(super) fn manifests<P: AsRef<Path>>(
     base_path: &P,
     metadata: Metadata,
 ) -> Result<Vec<ParsedManifest>, anyhow::Error> {
-    let mut packages = metadata
+    let mut packages: BTreeSet<(BTreeSet<Target>, PathBuf)> = metadata
         .workspace_packages()
         .iter()
         .copied()
         .chain(metadata.root_package())
-        .map(|p| (Some(p), p.manifest_path.clone().into_std_path_buf()))
-        .collect::<Vec<_>>();
+        .map(|p| {
+            (
+                gather_targets(p),
+                p.manifest_path.clone().into_std_path_buf(),
+            )
+        })
+        .collect::<BTreeSet<_>>();
```

Comment:
Re-reading it now, I think we can go one step further and make this a `BTreeMap<PathBuf, BTreeSet<Target>>`, which better conveys the actual intent/constraints here.

Reply:
Done.